### ISSUE TITLE
UTF-8 characters (when out of comments or quotes) are drawn with different font 

### ISSUE THREAD

**Issue by SevenLines** [sqlitebrowser/sqlitebrowser#364]
There is some old bug which presented from version 3.6. Look at this picture (first line) that's how it looks now, cyrillic characters (out of comments and quotes) are drawn with different font:
![image 10](https://cloud.githubusercontent.com/assets/5500999/8280145/0e3f9ea2-190d-11e5-84ce-87f600c57cda.png)

That's how it was in version 3.5.1 and probably the way it should be:
![image 6](https://cloud.githubusercontent.com/assets/5500999/8280115/c6f13772-190c-11e5-8a12-39bb50c75e8c.png)

There is the test string:
```SQL
абвгдеийклмнопрстуфхцчшщъыьэюя
-- абвгдеийклмнопрстуфхцчшщъыьэюя
SELECT *
FROM Student
WHERE name like 'Александр%'
```

**Comment by justinclift** [comment]
Interesting.  We added the ability to change the SQL font in a recent version (might have been 3.7.0, I don't remember).  I just tried setting the font to Andale Mono:
![screen shot 2015-06-22 at 14 33 03](https://cloud.githubusercontent.com/assets/406299/8283016/662f63a2-18ec-11e5-88dd-8d6eb36ebc76.png)
I think you are correct.  The grey text in the screenshot below is not the same size as the other text, and it should be: :frowning:

![screen shot 2015-06-22 at 14 33 06](https://cloud.githubusercontent.com/assets/406299/8282956/02df9a38-18ec-11e5-88b5-20f7bb26389a.png)

**Comment by SevenLines** [comment]
This problem is presented also on windows and linux. Also changing font size doesn't affect size of this characters. Looks like It's also affect arithmetic characters and digits:
![db browser for sqlite - -home-m-mmk-_akarin-lessons-lessons- -students dbs_058](https://cloud.githubusercontent.com/assets/5500999/8283570/383d5aa4-1922-11e5-9d4d-816f290e2124.png)

Maybe this is some special UTF8 digits, here the another test string:
```SQL
абвгдеийклмнопрстуфхцчшщъыьэюяqwerty
-- абвгдеийклмнопрстуфхцчшщъыьэюя
SELECT 1234567890-=/+,.\|
FROM Student 
WHERE name like 'Александр%'
```

**Comment by justinclift** [comment]
Yeah, this is definitely a bug of some sort.  I think @MKleusberg will be interested in this. :smile:

**Comment by MKleusberg** [comment]
This seems to be a bug in Scintilla or the QScintilla wrapper around it. I've just compiled the QScintilla example application (after enabling UTF-8 and SQL syntax highlighting in there) and the error still ocurrs. On the other hand, if I change the used lexer from SQL to C++ it seems to work - so the problem is probably somewhere in their SQL lexer and not in the general code, though I'm not entirely sure about that.

**Comment by MKleusberg** [comment]
Just for reference: The Qscintilla source code including the example application can be found here: http://www.riverbankcomputing.com/software/qscintilla/download
I've modified the QScintilla-gpl-2.9/example-Qt4Qt5/mainwindow.cpp file to enable UTF-8, include the lexer and use syntax highlighting; here's the new code: https://gist.github.com/MKleusberg/ceb2cf5f5ec84734b061

**Comment by justinclift** [comment]
@MKleusberg Would you be ok to ask on the QScintilla mailing list about it?

&nbsp; &nbsp; http://www.riverbankcomputing.com/mailman/listinfo/qscintilla

**Comment by SevenLines** [comment]
I have tried to compile binded example, changed font. 
```cpp
...
textEdit->setUtf8(true);
QsciLexerSQL *lex = new QsciLexerSQL(this);
lex->setFont(QFont("Courier New", 20));
textEdit->setLexer(lex);
...
```
This is what I got. All as it should be:
![untitled txt- - application_059](https://cloud.githubusercontent.com/assets/5500999/8285667/036b2aa6-192f-11e5-9854-ec85182bf5c8.png)

Some characters a bit bolder but all the rest is ok.

**Comment by SevenLines** [comment]
But setting font size to 12 looks a bit weird, maybe I do something wrong...
![untitled txt- - application_060](https://cloud.githubusercontent.com/assets/5500999/8285749/9ae9bb4a-192f-11e5-9d62-c3bf35e7d7de.png)

**Comment by justinclift** [comment]
So... what needs to happen next? :smile:

**Comment by SevenLines** [comment]
I found out that if make this small fix in SqlTextEdit::reloadSettings() function, everything starts to work, but I'm not sure will it brake something or not...
```cpp
...
defaultfont.setPointSize(PreferencesDialog::getSettingsValue("editor", "fontsize").toInt());
// sqlLexer->setDefaultFont(defaultfont); // comment this line
sqlLexer->setFont(defaultfont); // and replace with this line
setupSyntaxHighlightingFormat("comment", QsciLexerSQL::Comment);
...
```

**Comment by justinclift** [comment]
Want to make that a Pull Request? :smile:

**Comment by SevenLines** [comment]
Ok, let me try =)

**Comment by justinclift** [comment]
Closing this, as @SevenLines fixed it in #365. :smile:
